.pre-commit-config.yaml

```diff
@@ -1,3 +1,4 @@
+exclude: '^strictly_typed_pandas/_vendor'
```

Comment:
This part is not necessary anymore after your later commits, right?

Reply:
I think it's still useful; pre-commit will otherwise provide filenames explicitly to the linting commands, and some of them will in turn ignore their "excludes" rules because a filename was provided explicitly.

Reply:
I won't be able to write that documentation page for a few days at least, sorry.

I believe both the vendored and external import hooks works fine, but I have been just trusting the test suite for that. The only case where I see it doesn't work well is when the same module is processed twice by both the internal and external typeguard, which is why I blocked packages appearing (exactly) in both. Maybe there's a way to get around that (eg if the names aren't an exact match, but result in the same code being processed).

The other way it won't work is if an incompatible version of typeguard is installed and used, but the user was expecting strictly_typed_pandas to integrate with it. I believe this will silently not do what the user expects. We can raise a warning if an incompatible typeguard is installed and `--typeguard-packages` is used without `--stp-typeguard-packages`. The user can then silence that warning.

Reply:
> I won't be able to write that documentation page for a few days at least, sorry.

No problem, I'll write it!

> I believe both the vendored and external import hooks works fine, but I have been just trusting the test suite for that. The only case where I see it doesn't work well is when the same module is processed twice by both the internal and external typeguard, which is why I blocked packages appearing (exactly) in both. Maybe there's a way to get around that (eg if the names aren't an exact match, but result in the same code being processed).

I didn't mean the pytest extension actually, but rather this:
https://typeguard.readthedocs.io/en/stable/userguide.html#using-the-import-hook

The import hook will still work if the user has typeguard 2 installed. I'm fine with documenting that and then leaving support for the import hook in the vendored version for later.

> The other way it won't work is if an incompatible version of typeguard is installed and used, but the user was expecting strictly_typed_pandas to integrate with it. I believe this will silently not do what the user expects. We can raise a warning if an incompatible typeguard is installed and --typeguard-packages is used without --stp-typeguard-packages. The user can then silence that warning.

I think it's fine as it is. The warning would also only be thrown if stp is installed after typeguard, not the other way around. Let's leave it as it is.

What do you think?

Reply:
@willhardy I've added some documentation. What do you think?

If you agree, I'm happy to merge this!

Reply:
Nice documentation! covers everything, I believe :-)